packages/scxa-experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -0,0 +1,338 @@
+      selectedClusterIdOption: ``
+      }
+  }
+
+  render() {
+    const { location, match, history } = this.props
+    const { atlasUrl, suggesterEndpoint, defaultPlotMethodAndParameterisation } = this.props
+    const { species, experimentAccession, accessKey, ks, ksWithMarkerGenes, plotTypesAndOptions, metadata, anatomogram } = this.props
+    const search = URI(location.search).search(true)
+
+    let plotTypeDropdown =
+      Object.keys(defaultPlotMethodAndParameterisation)
+        .map(plot => ({
+          plotType: plot,
+          plotOptions: plotTypesAndOptions[plot]
+        }))
+
+    let organWithMostOntologies = Object.keys(anatomogram)[0]
+    for (let availableOrgan in anatomogram) {
+      organWithMostOntologies = anatomogram[availableOrgan].length > anatomogram[organWithMostOntologies].length ?
+       availableOrgan :
+       organWithMostOntologies
+    }
+
+    const routes = [
+      {
+        path: `/cell-plots`,
+        title: `Cell plots`,
+        main: () => <TSnePlotView
+          atlasUrl={atlasUrl}
+          suggesterEndpoint={suggesterEndpoint}
+          wrapperClassName={`row expanded`}
+          clusterPlotClassName={`small-12 large-6 columns`}
+          expressionPlotClassName={`small-12 large-6 columns`}
+          speciesName={species}
+          experimentAccession={experimentAccession}
+          accessKey={accessKey}
+          selectedPlotOption={search.plotOption || this.state.selectedPlotOption}
+          selectedPlotType={search.plotType || this.state.selectedPlotType}
+          ks={ks}
+          metadata={metadata.map(data => {return {value: data.value.replaceAll(`_`, ` `), label: data.label}})}
+          selectedColourBy={this.state.selectedColourByCategory === METADATA_PLOT ?
+              search.colourBy || this.state.selectedColourBy :
+              search.k || this.state.selectedClusterId
+          }
+          selectedColourByCategory={this.state.selectedColourByCategory} // Is the plot coloured by clusters or metadata
+          highlightClusters={search.clusterId ? JSON.parse(search.clusterId) : []}
+          geneId={search.geneId || ``}
+          height={800}
+          onSelectGeneId={
+            (geneId) => {
+              updateUrlWithParams([{geneId: geneId}])
+            }
+          }
+          plotTypeDropdown={plotTypeDropdown}
+          selectedPlotOptionLabel={search.plotOption ?
+                search.plotType ?
+                    getPlotOption(search.plotType.toLowerCase()) : getPlotOption(this.state.selectedPlotType.toLowerCase())
+                : this.state.selectedPlotOptionLabel
+          }
+          onChangePlotTypes={
+            (plotOption) => {
+              this.setState({
+                selectedPlotType: plotOption.value,
+                selectedPlotOption: Object.values(defaultPlotMethodAndParameterisation[plotOption.value])[0],
```

Comment:
Could we please extract `defaultPlotMethodAndParameterisation[plotOption.value]` to a local variable?
Thanks